pkgs/development/interpreters/python/hooks/default.nix

```diff
@@ -107,8 +107,11 @@ in {
     makePythonHook {
       name = "python-catch-conflicts-hook";
       substitutions = {
-        inherit pythonInterpreter pythonSitePackages;
-        catchConflicts=../catch_conflicts/catch_conflicts.py;
+        inherit pythonInterpreter pythonSitePackages setuptools;
```

Comment:
I think we can prevent many of the rebuilds by not inheriting setuptools here which also fixes  @vcunat concerns

Reply:
That is exactly what is required to import `pkg_resources`.

Reply:
I meant it needs to be put behind the same condition otherwise the env is for never python versions changed which triggers the rebuild.